source/agora/common/TransactionPool.d

```diff
@@ -274,11 +284,50 @@ public class TransactionPool
     {
         () @trusted
         {
-            auto results = this.db.execute("SELECT val FROM tx_pool");
+            // TXs that went through the "selection" process and should be
```

Comment:
Well this works, but I wonder if in the future we can replace it with something simpler. What I had in mind was:

- TransactionPool provides a range interface, maybe through a `.range()` getter or `opSlice`
- The element type of the range is a `Transaction[]`. So if you would call `.array()` on this range you would actually get `Transaction[][]`.
- If the length of an element of this range is == 1, it's not a double-spend and it can be selected.
- If the length of an element is > 1, it means this set of transactions are double-spends. The calling code can then decide what to do with it

Reply:
So, do we let double-spends through `opApply`?

Reply:
Well I would get rid of `opApply` entirely. But there might be use-cases for wanting to walk through double-spends in the future..

Reply:
Anyway it was just a suggestion, we can go ahead with your design for now.

Reply:
My intention was to keep the old behavior and API while extending the functionality.

Reply:
What you are describing is actually `spenders`. We can easily provide a mean to access it down the line, if we ever need that.